ext/mysql2/statement.c

```diff
@@ -204,6 +204,44 @@ static void set_buffer_for_string(MYSQL_BIND* bind_buffer, unsigned long *length
     xfree(length_buffers);                                  \
   }
 
+/* return 0 if the given bignum can cast as LONG_LONG, otherwise 1 */
+static int my_big2ll(VALUE bignum, LONG_LONG *ptr)
+{
+  unsigned LONG_LONG num;
+  size_t len;
+  int nlz_bits = 0;
+#ifdef HAVE_RB_ABSINT_SIZE
+  len = rb_absint_size(bignum, &nlz_bits);
+#else
+  len = RBIGNUM_LEN(bignum) * SIZEOF_BDIGITS;
+#endif
+  if (len > 8) goto overflow;
```

Comment:
Length 8 is intended to be the same as `sizeof(long long int)` right? Would it make sense for the integer size to be passed in as an argument?